### PR TITLE
Allow Labels to not maintain Order

### PR DESCRIPTION
**Pull request by benmcginnis**

### CODE REVIEW

File: location_diff.go [hunk 1]
```diff
@@ -51,7 +51,17 @@ func (y Location) Diff(b *Location) (d *Location, diff bool) {
 			continue
 		}
 
-		if !reflect.DeepEqual(valA.Interface(), valB.Interface()) {
+		aI, bI := valA.Interface(), valB.Interface()
+
+		unorderedA, aOk := aI.(*UnorderedStrings)
```
Comment (mhupman) [inline location_diff.go]: Shouldn't this just check if the object satisfies an Equal()/Compare() interface?  That way we don't have to add an exception for every type....

**Comment by mhupman** [comment]
Can we figure out how to share the compare logic between the multioption equals and this one?  Seems sad they are basically dupes of each other.

File: location.go [hunk 2]
```diff
@@ -467,3 +467,30 @@ type HolidayHours struct {
 	Date  string `json:"date"`
 	Hours string `json:"hours"`
 }
+
+// UnorderedStrings masks []string properties for which Order doesn't matter, such as LabelIds
+type UnorderedStrings []string
+
+// Equal implements compares UnorderedStrings
```
Comment (mhupman) [inline location.go]: grammar

File: location_diff.go [hunk 3]
```diff
@@ -51,7 +55,17 @@ func (y Location) Diff(b *Location) (d *Location, diff bool) {
 			continue
 		}
 
-		if !reflect.DeepEqual(valA.Interface(), valB.Interface()) {
+		aI, bI := valA.Interface(), valB.Interface()
+
+		comparableA, aOk := aI.(Comparable)
+		comparableB, bOk := bI.(Comparable)
+
+		if aOk && bOk {
```
Comment (mhupman) [inline location_diff.go]: Can we just inline these checks?  The number of local scoped variables with A and B is confusing

Comment (benmcginnis) [inline location_diff.go]: I don't think the magical inlining works with two type checks:
```
comparableA, aOk, comparableB, bOk := aI.(Comparable), bI.(Comparable)
```
gives "assignment count mismatch"

and this
```
if comparableA, aOk:= aI.(Comparable); comparableB, bOk := bI.(Comparable); aOk && bOk {
```
gives `syntax error missing { after if clause`

**Pull request merged**